### ISSUE TITLE
Option to unlock vault via device not shown after locking

### ISSUE THREAD

**Issue by septatrix** [bitwarden/clients#7805]
### Steps To Reproduce

1. Log in to the web vault (by any means).
2. Manually lock the vault.

### Expected Result

The option to unlock the vault with a device should be offered to unlock the vault (as it already does on an initial login).

### Actual Result

Only the option to unlock the vault with the master password is shown instead of alternative options like using a device.

### Screenshots or Videos

Initial login:
![image](https://github.com/bitwarden/clients/assets/24257556/af67d8cc-0be4-417e-b02a-a18a7e273442)

When unlocking a vault after it got locked:
![image](https://github.com/bitwarden/clients/assets/24257556/fa18d0a9-3d8e-4a75-a538-b9c90e30f07a)

### Additional Context

This is a refiling of #7629 which got closed under a false assumption. "Log in with device" is *not* limited to only the sign in process but also encompasses unlocking the vault! This is also what happens on a fresh login - it not only signs one in but also unlocks the vault.
Currently this is also my workaround: Instead of using my master password I simply log out completely and use the "Sign in using device" option.

I have no problem with this being closed as a duplicate but in that case please reopen the original issue (@SergeantConfused).

Furthermore this happens in the web vault, the browser extension and the desktop app.

### Operating System

Linux

### Operating System Version

Fedora 39

### Web Browser

Firefox

### Browser Version

121.0.1

### Build Version

2024.0.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by SergeantConfused** [comment]
Hi @septatrix,

Thank you for following up on this. I understand how this may appear since the act of logging in, in certain configurations, would also decrypt the vault in a similar manner to unlocking a logged-in client; However, those are two separate procedures with different supported methods of action. For example, you can log in via Enterprise Single Sign-On (SSO) using a [Trusted Device](https://bitwarden.com/help/setup-sso-with-trusted-devices/), whereas you can unlock a logged-in Bitwarden client using a PIN or a biometric sensor; Also, an internet connection is required in order to log in, whilst unlocking a logged-in Bitwarden client can be done offline.

In summary, unlocking a logged-in Bitwarden client is not the intended behaviour of the (Login with Device) function, and that is why it's not present in that flow in any Bitwarden client and why it would be considered a new feature. If you'd like to be able to use a device to unlock a logged-in Bitwarden client, similarly to how you'd log in from scratch using that function, please feel free to join the Bitwarden community and to vote for [this](https://community.bitwarden.com/t/unlock-with-device/51893) feature request.

I hope you find this clear and helpful, and I thank you in advance for your understanding,

**Comment by septatrix** [comment]
Thanks, that explanation makes it a bit more clear. To me it seemed very much like a bug that this is not offered natively but can be achieved using the rather janky workaround of logging out and back in using a device. Also thanks for linking the feature request in the forum - I voted for it and will continue following up there.
